selfdrive/car/mazda/mazdacan.py

```diff
@@ -117,3 +126,96 @@ def create_button_cmd(packer, car_fingerprint, counter, button):
     }
 
     return packer.make_can_msg("CRZ_BTNS", 0, values)
+
+
+def create_acc_cmd(self, packer, CS, CC, hold, resume):
+  ret = []
+  cp = CS.cp
+  cp_cam = CS.cp_cam
+  cc = CC
+  accel = CC.actuators.accel
+
+  if self.CP.carFingerprint in GEN1:
+    bus = 0
+    if cc.longActive:
+      accel = accel * 1170
+      accel = accel if accel < 1000 else 1000
+    else:
+      accel = int(cp_cam.vl["CRZ_INFO"]["ACCEL_CMD"]) # pass through
+    
+    values_21B = {
+        "ACC_ACTIVE"        : int(cc.longActive),
+        "ACC_SET_ALLOWED"   : int(bool(int(cp.vl["GEAR"]["GEAR"]) & 4)), # we can set ACC_SET_ALLOWED bit when in drive. Allows crz to be set from 1kmh.
+        "CRZ_ENDED"         : 0, # this should keep acc on down to 5km/h on my 2018 M3
+        "ACCEL_CMD"         : accel,
+        "STATIC_1"          : int(cp_cam.vl["CRZ_INFO"]["STATIC_1"]), #0x7FF,
+        "STATUS"            : int(cp_cam.vl["CRZ_INFO"]["STATUS"]),    #1
+        "MYSTERY_BIT"       : int(cp_cam.vl["CRZ_INFO"]["MYSTERY_BIT"]),
+        "CTR1"              : int(cp_cam.vl["CRZ_INFO"]["CTR1"])
+    }
+
+    values_21C = {
+        "CRZ_ACTIVE"       : int(cc.longActive),
+        "CRZ_AVAILABLE"    : int(cp_cam.vl["CRZ_CTRL"]["CRZ_AVAILABLE"]),
+        "DISTANCE_SETTING" : int(cp_cam.vl["CRZ_CTRL"]["DISTANCE_SETTING"]),
+        "ACC_ACTIVE_2"     : int(cc.longActive),
+        "DISABLE_TIMER_1"  : 0,
+        "DISABLE_TIMER_2"  : 0,
+        "NEW_SIGNAL_1"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_1"]),
+        "NEW_SIGNAL_2"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_2"]),
+        "NEW_SIGNAL_3"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_3"]),
+        "NEW_SIGNAL_4"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_4"]),
+        "NEW_SIGNAL_5"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_5"]),
+        "NEW_SIGNAL_6"     : int(cp_cam.vl["CRZ_CTRL"]["NEW_SIGNAL_6"]),
```

Comment:
Usually it's standard practice to copy the whole message in CarState, then start the dictionary at the copied values.

Like: https://github.com/commaai/openpilot/blob/master/selfdrive/car/hyundai/carstate.py#L145

Reply:
Thank you. I will clean that up.